pg/scripts/init.sql

```diff
@@ -52,61 +49,62 @@ create table reactions (
     sabio_rk text
 );
 
+create table organisms (
+    kegg text primary key not null,
+    entry text,
+    name text,
+    taxa text
+);
+
+create table domains (
+    abbreviation text primary key not null,
+    name text
+);
 
+create table genes (
+    kegg text,
```

Comment:
How come this one is not a `primary key`? It seems like there is no indexing at all now for the `genes` table. I feel like that would negatively impact the performance.

Reply:
Ah this is a tricky one, I forgot to document it.

Making `genes` a primary key would also make them unique. Currently this is a problem because of the `supplementary/gene/` files, since they end with something that is _like_ an empty line (say in VSCcode) but _not_ an empty line that can be replaced with `gsed` or `sed`. Even if those lines are removed, then the `cat` command joins the last line from the first file with the first one from the second. And for some reason `cat -s` didn't work either.

Reply:
I think there are some other problems with the `genes` data too. It seems that the `kegg` value have duplicates. For example:

<img width="311" alt="image" src="https://user-images.githubusercontent.com/423498/183395949-752ac221-c6b6-4989-a89d-add501140890.png">

When searching this looks like:

<img width="925" alt="image" src="https://user-images.githubusercontent.com/423498/183396053-d045475b-cc02-46b5-b158-f1b04edb2f2a.png">

And then the details page of course wouldn't load because we throw an error if the count is not 1.

I think we would want to consolidate these so all of the `ncbigene` and `uniprot` values are semicolon separated. Do you think that's reasonable or is there something else wrong or we should consider here?

Reply:
Totally my bad here - there was some merging being done before, for example in the first screenshot above - but that was done _per organism_, leading to the noticed duplicates. Consolidating would probably mean that ta new column for organisms would be added.

Since this is not straightforward to solve, I'm up for creating a different issue, to also handle the UI aspects of the refactoring (such as the links to the organism pages).

Reply:
Created a [new issue (bug)](https://github.com/MetabolicAtlas/MetabolicAtlas/issues/1067) where we can look more into this.